### PR TITLE
feat: cleanup testkube charts

### PR DESCRIPTION
**Pull request by dejanzele**
## Pull request description 
Cleanup of charts

## Checklist (choose whats happened)

- [ ] breaking change! (describe)
- [x] tested locally
- [x] tested on cluster
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

## Breaking changes

-

## Changes

- removed references to demo.testkube.io
- commented out `tls` and `hosts` config with examples in comment
- cleanup of `testkube` chart

## Fixes

- fixed a bug where `tlsenabled` was set as a string (`"false"`) but in the templates it was evaluated as a bool, so incorrect behaviour might occur
- fixed a bug where `testkube-api` Helm chart was missing `prometheus` block config and could not be installed independently

### CODE REVIEW

**Comment by dejanzele** [comment]
> values-demo.yaml and other values files are used in our cluster deployments, we cannot just remove them

But should we keep our internal deployment values in the public Helm chart for TestKube?
Same question for referencing local charts, should it be commited in our public Helm chart as is?

@TheBrunoLopes I'd like your opinion on this.

**Comment by vsukhin** [comment]
Sounds like it's time to introduce private Helm chart repo. But it should be a part of workflow @ypoplavs

**Comment by dejanzele** [comment]
@vsukhin this chart is already good, I'd definitely move environment-specific values files (demo, dev) out of this repo as there is no reason for end-users (or general public) to be aware of any of our internal deployments.

File: charts/testkube/Chart.yaml [hunk 1]
```diff
@@ -6,8 +6,8 @@ version: 1.5.60
 dependencies:
   - name: testkube-operator
     version: 1.5.1
-    #repository: https://kubeshop.github.io/helm-charts
-    repository: "file://../testkube-operator"
+    repository: https://kubeshop.github.io/helm-charts
+    #repository: "file://../testkube-operator"
```
[the remark this answers is not in the document]
Comment (dejanzele) [reply charts/testkube/Chart.yaml]: reverted

**Comment by TheBrunoLopes** [comment]
I'm adding a few more info on why this is very important that the charts are clean.
We should look at this repository as a public repository to be used by ANY user and for the community that wants to deploy Testkube, we should maximize compatibility. 

> values-demo.yaml and other values files are used in our cluster deployments, we cannot just remove them

**It's specially because that the value files are used in our clusters that we MUST remove them.**

Look at our team as users of this chart as any other organization. This chart is to be compatible with what the community uses foremost.

Would we approve a PR of a user that wants to add their values files for their clusters as well ? The answer is No right? 
Please look at this example of what is a public chart reposity, there's no values files that is only to be used by the maintainers. https://github.com/bitnami/charts/tree/master/bitnami

| Sounds like it's time to introduce private Helm chart repo

I'm not opposed to this idea, but before we do it we need to ask.. will the users also need to use a private repo to deploy their testkube installation ? If the answer is No, why is the chart good enough for them and not for us? If the answer is Yes. What is it that requires a private repo to be created for each user and not to use the public one ? is it something related with how people usually deploy helm charts ?

**Comment by exu** [comment]
I would approve this, but the issue is that if we'll merge it it'll brake our demo as helm chart will be automatically deployed without ingress

So please make all move those files first to some repo and merge after pipelines will be managed.

**Comment by ypoplavs** [comment]
I second your opinion @TheBrunoLopes, but first we need to move the deployment process to another repo and then remove the values file for `demo` and `stage` otherwise it will break it.

**Pull request closed without merging**